### DISCUSSION TITLE
session  만료 시간에 대한 의견

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-teams/2023-pium#244] [뜨거운 토론]
## 주제
로그인 session 만료 시간을 설정하고자 하여, 이에 대한 여러분의 의견을 알아보고자 합니다!

 세션 유지 시간이 짧을 경우 보안 및 성능상 이점이 있을 수 있지만, 사용자의 불편함이 증가할 수 있습니다. 
유지 시간이 길 경우에는 반대의 특성을 가질 수 있겠네요!

세션 유지 시간에 대한 생각이 모두 다를 수 있기 때문에, 모두의 의견을 통해 결정하고자 합니다.

`적절한 세션 유지 시간`에 대한 각자의 의견을 댓글로 달아주시면 되겠습니다!

**Comment by Choi-JJunho** [comment]
저는 세션을 사용한다면 6시간도 적절하다고 생각합니다.
로그인 유지시간을 짧게 가져가는 이유는 사용자 식별값이 탈취되는 상황을 우려하기 때문이라고 생각합니다.
하지만 해당 이유는 무상태성을 가지는 JWT를 사용하는 상황에서 탈취된 토큰에 대해 서버가 대처할 수 없기 때문에 최우선 적으로 조치해야하는 내용이라고 생각합니다.

서버에서 사용자의 로그인 정보를 들고있는 세션의 경우 사용자가 로그아웃요청 혹은 서버에서 강제 만료시킬 수 있는 방안이 존재하기 때문에 세션의 유지 시간이 길다는 특징으로부터 발생하는 단점(탈취문제)은 최소화되었다고 생각합니다.

리마인더라는 기능을 사용하기 위해 웹으로 한번 오는 것 조차 번거로운데 짧은 시간내로 로그인마저 풀리게된다면 사용자 경험 측면에서 좋지 못할 것 같기 때문에 저는 `세션 만료시간은 6시간`이라는 의견입니다

**Comment by hozzijeong** [comment]
궁금한게 있는데, 세션만료가 되면 다시 로그인을 해야하나요? 자동 갱신할 수 있는 방법은 없나요?

**Reply by Choi-JJunho** [reply]
만료가되면 다시 로그인을 해야할 것 같아요!

자동갱신은 지금 생각나는 방식으로는 두가지 방식이 생각나네요!

1. 사용자가 로그인 연장 버튼을 누르도록 한다.
2. 사용자의 요청이 들어올 때 마다 세션을 갱신한다.

**Reply by hozzijeong** [reply]
고민이네요... 세션 유효시간이 30분이나 1시간이라면 수동 갱신이 괜찮은데, 6시간 이상 가져가면 굳이 수동 갱신이 필요할까 하는 생각이 듭니다.

두 번째의 경우에는, 결국 해당 세션이 "유효한 세션"일 경우에만 가능하다고 생각이 들어서, 만료된 세션임에도 자동으로 로그인을 유지할 수 있는 방법은 아닌것 같습니다. 

저도 한 6시간이 적당한것 같아요 (카카오 Access Token의 유효시간도 6시간으로 알고잇습니다 ㅎ)

**Comment by rawfishthelgh** [comment]
저는 세션 시간을 길게 가져가는 것에 동의합니다!

순수히 사용자의 편의성 측면에서 생각하자면 
여러 개의 식물을 관리하고 있는 사용자가 존재하는데, 
각 식물을 관리하는 시간 도중 로그인이 풀린다면 불편함이 생길 것 같네요.

궁금한 것은 `세션 시간이 길다`라면 보통 몇 시간 이상이고, `보통` 혹은 `짧다`라고 했을 때 몇 시간 이하가 될까요??

**Reply by yeonkkk** [reply]
명확하게 정해져 있는 건 아닌 것 같아요.
근데 세션 유효 시간이라는게 `마지막 요청으로부터 설정한 시간 까지 세션이 유효`한 거라서 저는  1시간 정도가 보통이고 30분 정도가 짧은 것 이라고 생각했습니다ㅎ

**Comment by WaiNaat** [comment]
탈취의 위협이 크지 않다면 길게 가져가는 게 좋을 것 같아요!
별개로 각종 사이트에서 며칠이 지나도 로그인이 유지되는 거는 다른 기법을 추가적으로 사용해야 하는 건가요?

**Reply by Choi-JJunho** [reply]
로그인 유지같은 체크박스를 만들어서 만료기간이 없는 세션ID를 발급할 수도 있을 것 같네요

**Reply by Choi-JJunho** [reply]
세션 만료기간이 진짜 3~4일이고 접속할 때 마다 갱신된다면 참새가 말한 시나리오를 비슷하게 가져갈 수도 있을 것 같네요

**Comment by yeonkkk** [comment]
모두 의견 감사합니다 ~ 우선 6시간으로 적용하겠습니다!
